pkg/manager/integration/manager_test.go

```diff
@@ -0,0 +1,286 @@
+		mgr, err := manager.New(env.Config, manager.Options{
+			Scheme:                 scheme,
+			HealthProbeBindAddress: ":0",
+			// Disable metrics to avoid port conflicts.
+			MetricsBindAddress: "0",
+			WebhookServer: webhook.NewServer(webhook.Options{
+				Port:    env.WebhookInstallOptions.LocalServingPort,
+				Host:    env.WebhookInstallOptions.LocalServingHost,
+				CertDir: env.WebhookInstallOptions.LocalServingCertDir,
+			}),
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Configure health probes.
+		Expect(mgr.AddReadyzCheck("webhook", mgr.GetWebhookServer().StartedChecker())).To(Succeed())
+		Expect(mgr.AddHealthzCheck("webhook", mgr.GetWebhookServer().StartedChecker())).To(Succeed())
+
+		// Set up Driver reconciler.
+		driverReconciler := &DriverReconciler{
+			Client: mgr.GetClient(),
+		}
+		Expect(ctrl.NewControllerManagedBy(mgr).For(&crewv1.Driver{}).Complete(driverReconciler)).To(Succeed())
+
+		// Set up a conversion webhook.
+		conversionWebhook := createConversionWebhook(mgr)
+		mgr.GetWebhookServer().Register("/convert", conversionWebhook)
+
+		// Add an index on v2 Driver.
+		Expect(mgr.GetCache().IndexField(ctx, &crewv2.Driver{}, "name", func(object client.Object) []string {
+			return []string{object.GetName()}
+		})).To(Succeed())
+
+		// Start the Manager.
+		ctx, cancel := context.WithCancel(ctx)
+		go func() {
+			defer GinkgoRecover()
+			Expect(mgr.Start(ctx)).NotTo(HaveOccurred())
+		}()
+
+		// Verify manager.Start successfully started health probes, webhooks, caches (including informers) and reconcilers.
+		<-mgr.Elected()
+
+		// Verify the reconciler reconciles.
+		Eventually(func(g Gomega) {
+			g.Expect(atomic.LoadUint64(&driverReconciler.ReconcileCount)).Should(BeNumerically(">", 0))
+		}, 10*time.Second).Should(Succeed())
+
+		// Verify conversion webhook was called.
+		Expect(atomic.LoadUint64(&conversionWebhook.ConversionCount)).Should(BeNumerically(">", 0))
+
+		// Verify the conversion webhook works.
+		driverV2 := &unstructured.Unstructured{}
+		driverV2.SetGroupVersionKind(crewv2.GroupVersion.WithKind("Driver"))
+		driverV2.SetName("driver1")
+		driverV2.SetNamespace(metav1.NamespaceDefault)
+		Expect(c.Get(ctx, client.ObjectKeyFromObject(driverV2), driverV2)).To(Succeed())
+
+		// Shutdown the server
+		cancel()
+	})
+})
+
+type DriverReconciler struct {
+	Client         client.Client
+	ReconcileCount uint64
+}
+
+func (r *DriverReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("Reconciling")
+
+	// Fetch the Driver instance.
+	cluster := &crewv2.Driver{}
+	if err := r.Client.Get(ctx, req.NamespacedName, cluster); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	atomic.AddUint64(&r.ReconcileCount, 1)
+
+	return reconcile.Result{}, nil
+}
+
+type ConversionWebhook struct {
```

Comment:
Is this just a shim on the conversion handler in the webhook package? Could we add a comment to the struct to that effect?